src/data/option/basic.lean

```diff
@@ -119,6 +119,106 @@ lemma comp_map (h : β → γ) (g : α → β) (x : option α) :
   option.map g ∘ option.map f = option.map (g ∘ f) :=
 by { ext x, rw comp_map }
 
+lemma mem_map_of_mem {α β : Type*} {a : α} {x : option α} (g : α → β) (h : a ∈ x) : g a ∈ x.map g :=
+mem_def.mpr ((mem_def.mp h).symm ▸ map_some')
+
+section pmap
+
+variables {p : α → Prop} {x : option α} {f : Π (a : α), p a → β}
+
+lemma pbind_eq_none {f : Π (a : α), a ∈ x → option β}
+  (h' : ∀ a ∈ x, f a H = none → x = none) :
+  x.pbind f = none ↔ x = none :=
+begin
+  cases x,
+  { simp },
+  { simp only [pbind, iff_false],
+    intro h,
+    specialize h' x rfl h,
+    contradiction }
+end
+
+lemma pbind_eq_some {f : Π (a : α), a ∈ x → option β} {y : β}
+   :
+  x.pbind f = some y ↔ ∃ (z ∈ x), f z H = some y :=
+begin
+  cases x,
+  { simp },
+  { simp only [pbind],
+    split,
+    { intro h,
+      use x,
+      simpa only [mem_def, exists_prop_of_true] using h },
+    { rintro ⟨z, H, hz⟩,
+      simp only [mem_def] at H,
+      simpa only [H] using hz } }
+end
+
+@[simp] lemma pbind_eq_bind {f : α → option β} :
```

Comment:
```suggestion
@[simp] lemma pbind_eq_bind (f : α → option β) :
```
`f` is not inferable from later arguments, is there a reason for it to be implicit here?

Reply:
Maybe I'm missing something, but isn't it true that we can make arguments implicit if they appear on both sides of an equality or iff, since we usually just use these lemmas in rewriting?

Reply:
Ah, sorry, I lost track of this in the craziness of the last few weeks. Yes, for using these in `rw`/`simp` it doesn't matter, but it does if you want to use them in e.g. `have`. For `iff`, the `lemma_name.mp` only works if all the arguments are implicit, which is a good reason to prefer that. But `eq.mp` is less commonly used. That said it's not so important.